### PR TITLE
Move WildFly Preview to a native jakarta namespace variant of Hibernate ORM 6.0.0.Final

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.redhat.com/browse/WFLY-15440
https://issues.redhat.com/browse/WFLY-15694
https://issues.redhat.com/browse/WFLY-16178

Signed-off-by: Scott Marlow <smarlow@redhat.com>

### CODE REVIEW

**Comment by scottmarlow** [comment]
`ScannerTests` passed this time but now seem to fail on `compat` tests:
```
[ERROR] /store/work/tc-work/e34a6f994de9f7c6/testsuite/compat/src/test/scripts/build-jars.xml:28: Warning: Could not find file /store/work/tc-work/e34a6f994de9f7c6/testsuite/compat/${org.hibernate.common:hibernate-commons-annotations:jar} to copy.
[14:09:44]
[Step 2/3] [ERROR] around Ant part ...<ant antfile="src/test/scripts/build-jars.xml">... @ 6:50 in /store/work/tc-work/e34a6f994de9f7c6/testsuite/compat/target/antrun/build-main.xml
[14:09:44]
[Step 2/3] [ERROR] -> [Help 1]
```

**Comment by scottmarlow** [comment]
The compat test failure recreates for me locally.  If we merge https://github.com/wildfly/wildfly/pull/14882, then that will address the failure.

If we decide to not merge https://github.com/wildfly/wildfly/pull/14882, we could evaluate which Hibernate ORM version the compat tests should use and how.

**Comment by scottmarlow** [comment]
Attached is a summary of Hibernate ORM 6 failures that I was able to recreate locally.  [orm6cr1.txt](https://github.com/wildfly/wildfly/files/7995175/orm6cr1.txt)

File: jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/VirtualFileSystemArchiveDescriptorFactory.java
Comment (boris-unckel) [inline jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/VirtualFileSystemArchiveDescriptorFactory.java:36]: Maybe it's intended: A factory with an static instance but the constructor is default and therefor public. Is a private constructor needed here?

File: jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/HibernateManagementAdaptor.java
Comment (boris-unckel) [inline jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/HibernateManagementAdaptor.java:32]: Maybe it's intended: A singleton with an static instance but the constructor is default and therefor public. Is a private constructor needed here?

File: jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/QueryName.java
Comment (boris-unckel) [inline jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/QueryName.java:114]: Maybe it's intended: A static factory method but the constructor is public. Is a private constructor needed here?

File: jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/VirtualFileSystemArchiveDescriptorFactory.java
Comment (scottmarlow) [reply jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/VirtualFileSystemArchiveDescriptorFactory.java:39]: Thanks @boris-unckel , just switched to a private constructor

File: jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/HibernateManagementAdaptor.java
Comment (scottmarlow) [reply jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/HibernateManagementAdaptor.java:32]: Thanks @boris-unckel , just switched to a private constructor

File: jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/QueryName.java
Comment (scottmarlow) [reply jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/management/QueryName.java]: Thanks @boris-unckel , just switched to a private constructor

**Comment by bstansberry** [comment]
@scottmarlow I disabled the compat tests from -DallTests until we can get #14882 merged.

**Comment by bstansberry** [comment]
@scottmarlow https://github.com/wildfly/wildfly/pull/15236 should help with one or more failures in the WildFly Preview CI job.

**Comment by bstansberry** [comment]
Also https://github.com/wildfly/wildfly/pull/15237

**Comment by bstansberry** [comment]
@scottmarlow I merged #15236 so I kicked off the WildFly Preview job again to see what effect that had.

**Comment by bstansberry** [comment]
I merged #15237 so I kicked off another run of the WildFly Preview CI job to see the results (https://ci.wildfly.org/viewLog.html?buildId=296280&buildTypeId=WF_PullRequest_WildFlyPreviewLinuxJdk17). When it's done the status box here will update.

**Comment by scottmarlow** [comment]
Pending the merge of https://github.com/wildfly/wildfly/pull/15375, hibernate6/src/main/java/org/jboss/as/jpa/hibernate/ManagedEmbeddedCacheManagerProvider.java should also be updated to:
```
if (manager.getCacheManagerConfiguration().statistics()) {
                settings.setProperty(STATISTICS, Boolean.TRUE.toString());
            }
```

**Comment by scottmarlow** [comment]
FYI @pferraro https://github.com/wildfly/wildfly/pull/15165/commits/5f8e5507b61a22f91272be972bedfb18a0df549a applies your https://github.com/wildfly/wildfly/pull/15375 change to Hibernate ORM 6 integration.

**Comment by scottmarlow** [comment]
I expect that https://github.com/wildfly/wildfly/pull/15185 will resolve the Hibernate Search test failure.

**Comment by scottmarlow** [comment]
> I expect that #15185 will resolve the Hibernate Search test failure.

And it does!

**Comment by scottmarlow** [comment]
Synced with latest `main` branch, no changes have been made otherwise.

**Comment by scottmarlow** [comment]
Soon we can look at switching to ORM 6.0.1.

File: ee-9/common/pom.xml
Comment (bstansberry) [inline ee-9/common/pom.xml:632]: Please add an entry for this to https://github.com/wildfly/wildfly/blob/main/ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml

File: ee-9/common/pom.xml
Comment (bstansberry) [inline ee-9/common/pom.xml:637]: Please add an entry for this to https://github.com/wildfly/wildfly/blob/main/ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml

File: ee-9/common/pom.xml
Comment (bstansberry) [inline ee-9/common/pom.xml:824]: Please add an entry for this to https://github.com/wildfly/wildfly/blob/main/ee-9/common/src/main/resources/license/preview-feature-pack-common-licenses.xml

File: ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/antlr/main/module.xml
Comment (bstansberry) [inline ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/antlr/main/module.xml:1]: Please move this file to https://github.com/wildfly/wildfly/tree/main/ee-9/common

File: ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/6.0/module.xml
Comment (bstansberry) [inline ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/6.0/module.xml:1]: Please move this file to https://github.com/wildfly/wildfly/tree/main/ee-9/common

File: ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/jipijapa-hibernate6/main/module.xml
Comment (bstansberry) [inline ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/jipijapa-hibernate6/main/module.xml:1]: Please move this file to https://github.com/wildfly/wildfly/tree/main/ee-9/common

File: ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml
Comment (bstansberry) [inline ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/main/module.xml:1]: Please move this file to https://github.com/wildfly/wildfly/tree/main/ee-9/common

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/Hibernate2LCacheStatsTestCase.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/Hibernate2LCacheStatsTestCase.java:95]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/Hibernate2LCacheStatsTestCase.java:95]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/Hibernate4NativeAPIProviderTestCase.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/Hibernate4NativeAPIProviderTestCase.java:84]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/HibernateNativeAPITransactionTestCase.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/HibernateNativeAPITransactionTestCase.java:84]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/envers/Hibernate4NativeAPIEnversTestCase.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/envers/Hibernate4NativeAPIEnversTestCase.java:83]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/naturalid/HibernateNativeAPINaturalIdTestCase.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/naturalid/HibernateNativeAPINaturalIdTestCase.java:98]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/secondlevelcache/HibernateSecondLevelCacheTestCase.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/secondlevelcache/HibernateSecondLevelCacheTestCase.java:89]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/entitytest/EntityTestCase.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/entitytest/EntityTestCase.java:78]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/sessionfactorytest/SessionFactoryTestCase.java
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/sessionfactorytest/SessionFactoryTestCase.java:61]: You need to check for ts.bootable.ee9 as well as that enables the profile used in jobs like https://ci.wildfly.org/viewType.html?buildTypeId=WF_MainNightlyJobs_WildFlyPreviewBootableJarLinuxJdk11

**Review by bstansberry — CHANGES_REQUESTED** [review]
@scottmarlow I have a bunch of comments here, but really it's just a few simple things, a couple of which apply to multiple files.

Note that there are also a couple of test failures in the WildFly Preview job.

I didn't really look at what's in the jpa/hibernate6 module. I'll poke at that a bit, mostly out of curiousity, but wanted to get you this feedback quickly.

File: ee-9/common/pom.xml
Comment (scottmarlow) [reply ee-9/common/pom.xml:632]: using:
```
          <name>The Antlr4 4.9.1 License</name>
          <url>https://github.com/antlr/antlr4/blob/4.9.1/LICENSE.txt</url>
          <distribution>repo</distribution>
```

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/sessionfactorytest/SessionFactoryTestCase.java
Comment (scottmarlow) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/sessionfactorytest/SessionFactoryTestCase.java]: Thanks Brian, added!

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/entitytest/EntityTestCase.java
Comment (scottmarlow) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/entitytest/EntityTestCase.java:78]: Thanks Brian, added!

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/naturalid/HibernateNativeAPINaturalIdTestCase.java
Comment (scottmarlow) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/naturalid/HibernateNativeAPINaturalIdTestCase.java]: Thanks Brian, added!

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/envers/Hibernate4NativeAPIEnversTestCase.java
Comment (scottmarlow) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/hibernate/envers/Hibernate4NativeAPIEnversTestCase.java]: Thanks Brian, added!

File: ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/6.0/module.xml
Comment (scottmarlow) [reply ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/hibernate/6.0/module.xml]: Thanks Brian, moved!

File: ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/antlr/main/module.xml
Comment (scottmarlow) [reply ee-9/feature-pack/src/main/resources/modules/system/layers/base/org/antlr/main/module.xml:1]: Thanks Brian, moved!

File: ee-9/common/pom.xml
Comment (scottmarlow) [reply ee-9/common/pom.xml:835]: Thanks Brian, entry added!

File: ee-9/common/pom.xml
Comment (scottmarlow) [reply ee-9/common/pom.xml:648]: Thanks Brian, entry added!

File: ee-9/common/pom.xml
Comment (scottmarlow) [reply ee-9/common/pom.xml:643]: Thanks Brian, entry added!

**Comment by scottmarlow** [comment]
> Note that there are also a couple of test failures in the WildFly Preview job.

Hmm, I will look at org.jboss.as.test.integration.jca.statistics.xa.XaDataSourcePoolStatisticsTestCase, didn't see that before.

The Search test failure will be addressed by https://github.com/wildfly/wildfly/pull/15185

**Comment by bstansberry** [comment]
@scottmarlow I suspect XaDataSourcePoolStatisticsTestCase can be fixed by changing the dialect to H2Dialect

https://github.com/wildfly/wildfly/blob/main/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/statistics/xa/persistence.xml#L31

It looks like Hibernate 6 is using SequenceInformationExtractorHSQLDBDatabaseImpl when HSQLDialect is used. Which seems right, given the persistence.xml says to use HSQLDialect. But SequenceInformationExtractorHSQLDBDatabaseImpl didn't exist in Hibernate 5.3.

If that works I don't know if that indicates a Hibernate 6 bug.

Re the Hibernate search test, is the plan to not merge this and instead pick it up in the #15182 PR? If not we can disable the test here and then re-enable in the Hibernate Search PR.

**Comment by scottmarlow** [comment]
> @scottmarlow I suspect XaDataSourcePoolStatisticsTestCase can be fixed by changing the dialect to H2Dialect
> 
Changing to H2Dialect doesn't seem to help.  I also tried removing the dialect from the persistence.xml, that didn't impact the test results.

> https://github.com/wildfly/wildfly/blob/main/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/statistics/xa/persistence.xml#L31
> 
> It looks like Hibernate 6 is using SequenceInformationExtractorHSQLDBDatabaseImpl when HSQLDialect is used. Which seems right, given the persistence.xml says to use HSQLDialect. But SequenceInformationExtractorHSQLDBDatabaseImpl didn't exist in Hibernate 5.3.
> 
> If that works I don't know if that indicates a Hibernate 6 bug.

From debugging, the org.jboss.as.test.integration.transactions.TransactionCheckerSingleton#addRollback is called 10 times but the test results don't reflect that for some reason.  

> 
> Re the Hibernate search test, is the plan to not merge this and instead pick it up in the #15182 PR? If not we can disable the test here and then re-enable in the Hibernate Search PR.

+1

**Comment by bstansberry** [comment]
@scottmarlow Changing to H2Dialect does make a difference though. The test still fails, but at least the test app can deploy properly.

Also, I merged something and introduced a conflict!

**Comment by scottmarlow** [comment]
> @scottmarlow Changing to H2Dialect does make a difference though. The test still fails, but at least the test app can deploy properly.

I pushed a change to remove the dialect in the test so that Hibernate will auto discover the dialect.  The test still fails for me the same.

> 
> Also, I merged something and introduced a conflict!

Resolved the conflict, thanks!

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/entitytest/EntityTestCase.java
Comment (scottmarlow) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/hibernate/entitytest/EntityTestCase.java:78]: Somehow I lost these changes, making them again via `assumeTrue(System.getProperty("ts.ee9") == null && System.getProperty("ts.bootable.ee9") == null)`

**Pull request merged**
